### DISCUSSION TITLE
Changing the WYSIWYG editor - is it possible? Why was CKEditor even chosen as it is GPL?

### DISCUSSION THREAD

**Discussion by Blackbam** [sulu/sulu#7690] [Q&A]
Hi! I am new to Sulu and played around with the demo a little bit, noticed that Sulu is using CKEditor in the backend. CKEditor is GPL instead of MIT which can cause significant licensing issues for some companies  (depending on the use case).

Is it complicated to change the WYSIWYG editor in Sulu for the content editing? Why did you even make that choice? CKEditor is a trap for future costs that may arise (even heard that they have a conglomerate which bought TinyMCE and has put it under strict licensing terms as well).

Here is a purely MIT licensed for of TinyMCE which would be great to use: https://github.com/hugerte/hugerte

If anyone has insights on the topic I would be glad to hear an answer, also if someone knows how complicated it is to replace the backend editor.

**Comment by alexander-schranz** [comment]
Hello @Blackbam,

We decided to use CKEditor because it fulfills all our requirements and offers a well-maintained, actively developed solution. There are no issues with the GPL license, as CKEditor allows its use within Sulu through a special agreement (Free for Open Source License Agreement). You can read more about it here: [Sulu Case Study](https://ckeditor.com/case-studies/sulu/). The same applies, for example, to [Neos CMS](https://ckeditor.com/case-studies/neos/).

If you still prefer to use your another text editor, you can create a new field and content type yourself. Keep in mind that you'll need an editor compatible with React and its behavior, so not every rich text editor will work seamlessly within it. More details can be found here: Sulu Demo PR https://github.com/sulu/sulu-demo/pull/79.

Best regards,
Alex

**Reply by Blackbam** [reply]
@alexander-schranz Thanks for the explanation! Got it. Does it mean I can use Sulu for commercial projects without restrictions? Or do I have conflicts with the CKEditor license than? I know for some projects it does not matter, but in some cases it might be very relevant?

**Reply by alexander-schranz** [reply]
Hello @Blackbam,

I'm not in the details of the License. But you can use Sulu without any issues to create custom Websites or Application for your clients like any other Web Agency using Sulu or similar other CMS Systems. What you are selling here is not a Software you sell the implementation of a Website / Application build ontop of Sulu / Symfony which is fine under the condition of the Free for Open Source Software License.

Another Example of GPL License is Contao CMS is licensed under GPL and WebAgencies can create website and applications with it for there clients without any issues. So it always depends on what you Sell. For typical client work where the CKEditor ist just part of Sulu CMS / Admin you don't need to care because of the FfOSS agreement to use Sulu.

If you create a general product yourself with the target of selling that commercial product to multiple clients I'm not sure if that still can be seen as part of that License. GPL mostly is there to avoid that somebody copy a software put another name on it and sell it where the most work is done by others.

To be clear the FfOSS agreement is for the usage of CKEditor inside Sulu Content Management System to manage your content, Sulu Objects, Custom entities, .... and more. If you create Forms on your Website or in a custom Symfony part outside the Sulu Admin and want to use there the CKEditor you would as I understand need to purchase a license I think from CKEditor.

Best Regards,
Alex

**Reply by Blackbam** [reply]
@alexander-schranz First of all: It is very good that Sulu is MIT licensed! 

Regarding the selling you are right, some companies have issues with GPL depending on the use case. They do not buy projects if GPL requirements are violated. Even WordPress is GPL and it powers half of the internet, but there you have all those discussion about themes and plugins for instance: Some say, those are "derivate works" and therefore must be free for all. It is not entirely clear in many cases.

So if the FfOSS agreement covers commercial usage for the backend in commercial closed-source projects that is enough for me, delivering CKEditor to frontend might not be covered but that is a rare use case. 

Usually, for backend usage the LGPL was invented but CKEditor has GPLv2 only so this FfOSS agreement has to state clearly that backend usage in commercial projects is allowed. Can I find it somewhere?

Thanks,
David

**Reply by alexander-schranz** [reply]
If you want to get a sure answer you need to get into contact with ckeditor directly, I'm not a Lawyer and wasn't involved at that Time CKEditor was added to Sulu. I don't know what you try to create when you say comercial closed source project. Creating website / application you not creating a commercial "product" what I will call it when you create software you try to sell to different companies. You are doing normal agency work and sell working hours not a product so you are like every other Typo3, Drupal, Neos CMS .. Agency, which all systems us CKEditor 5 on the free basis for there clients. If you are unsure about what you try to achieve check with CKEditor directly.

**Reply by pinkchilla** [reply]
I just saw this conversation and maybe somebody here has an answer to my question. Am I right that If I use the GPL CKEditor in my  otherwise proprietary software the editor with all it’s alterations must always be released under GPL whereas the rest of my software can remain proprietary?

**Reply by alexander-schranz** [reply]
Sorry, nothing to add here, the answer stays the same, ask CKEditor if you are unsure: https://github.com/sulu/sulu/discussions/7690#discussioncomment-11556043.
